Sources/zcash-swift-payment-uri/Model.swift

```diff
@@ -0,0 +1,245 @@
+//
+//  Model.swift
+//
+//
+//  Created by Pacu on 2023-11-07.
+//
+
+import Foundation
+
+struct PaymentRequest {
+    let payments: [Payment]
+}
+
+/// A Single payment that will be requested
+struct Payment {
+    /// Recipient of the payment.
+    let recipientAddress: RecipientAddress
+    /// The amount of the payment expressed in decimal ZEC
+    let amount: Amount
```

Comment:
If possible, this should be a nonnegative type; negative payment amounts don't make sense.

Reply:
Yes, the problem is that I'd have to duplicate the Zatoshi Implementation or importing the zcash-sdk as a dependency that probably, callers already have in the version of their choice.

Reply:
What I mean is, your `Amount` struct that's defined in this module should just be `Zatoshi`, and its value should be a nonnegative type. Zip 321 has no use anywhere for negative values (which are always value deltas). so such values should not be representable.

Reply:
thanks for elaborating on this. I added validation for that. I'm speculating on Zatoshi being of little use when actually creating a transaction request URI from a Wallet UI. So I prioritize a Decimal API first.